lib/chef/win32/security.rb

```diff
@@ -104,6 +104,22 @@ def self.add_access_denied_ace_ex(acl, sid, access_mask, flags = 0, revision = A
         end
       end
 
+      def self.add_account_right(name, privilege)
+        privilege_pointer = FFI::MemoryPointer.new LSA_UNICODE_STRING, 1
+        privilege_lsa_string = LSA_UNICODE_STRING.new(privilege_pointer)
+        privilege_lsa_string[:Buffer] = FFI::MemoryPointer.from_string(privilege.to_wstring)
+        privilege_lsa_string[:Length] = privilege.length * 2
```

Comment:
no. I fiddled with this quite alot. Even `privilege.to_wstring.size` is too small